src/test/java/seedu/clinic/logic/parser/ViewCommandParserTest.java

```diff
@@ -0,0 +1,95 @@
+    }
+
+    @Test
+    public void parse_onlyTypeNoKeywords_throwParsesException() {
+        assertParseFailure(parser, "supplier ", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_TOO_FEW_ARGUMENTS));
+        assertParseFailure(parser, "suPPLIEr ", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_TOO_FEW_ARGUMENTS));
+        assertParseFailure(parser, "WAREHOUSE ", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_TOO_FEW_ARGUMENTS));
+        assertParseFailure(parser, "WAREhouse ", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_TOO_FEW_ARGUMENTS));
+        assertParseFailure(parser, "warehouse ", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_TOO_FEW_ARGUMENTS));
+    }
+
+    @Test
+    public void parse_onlyOneKeywordNoType_throwParsesException() {
+        assertParseFailure(parser, "alice", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_TOO_FEW_ARGUMENTS));
+        assertParseFailure(parser, "bENson", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_TOO_FEW_ARGUMENTS));
+    }
+
+    @Test
+    public void parse_onlyMultipleKeywordsNoType_throwParsesException() {
+        assertParseFailure(parser, "alice benson", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_INVALID_TYPE));
+        assertParseFailure(parser, "ALIce bENson", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_INVALID_TYPE));
+    }
+
+    @Test
+    public void parse_wrongTypeCorrectMultipleKeyWords_throwParsesException() {
+        assertParseFailure(parser, "VIEWWArehouse alice benson", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_INVALID_TYPE));
+        assertParseFailure(parser, "VIEWSupplier ALIce bENson", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_INVALID_TYPE));
+        assertParseFailure(parser, "See ALIce bENson", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_INVALID_TYPE));
+    }
+
+    @Test
+    public void parse_wrongTypeCorrectOneKeyWord_throwParsesException() {
+        assertParseFailure(parser, "VIEWWArehouse alice", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_INVALID_TYPE));
+        assertParseFailure(parser, "VIEWSupplier bENson", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_INVALID_TYPE));
+        assertParseFailure(parser, "See ALIce", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_INVALID_TYPE));
+    }
+
+    @Test
+    public void parse_correctTypeCorrectKeyWords_success() {
+        ViewCommand expectedViewCommand1 = new ViewCommand("supplier", Arrays.asList(new String[]
```

Comment:
I think you can just use list.of() instead of arrays.aslist(String[]{})

Reply:
Thanks for the suggestion, let me try to use that instead of arrays.aslist(String[]{})